src/wallet/rpcwallet.cpp

```diff
@@ -3281,7 +3290,23 @@ UniValue signrawtransactionwithwallet(const JSONRPCRequest& request)
     // Parse the prevtxs array
     ParsePrevouts(request.params[1], nullptr, coins);
 
-    return SignTransaction(mtx, &*pwallet->GetLegacyScriptPubKeyMan(), coins, request.params[2]);
+    std::set<const SigningProvider*> providers;
+    for (const std::pair<COutPoint, Coin> coin_pair : coins) {
+        const SigningProvider* provider = pwallet->GetSigningProvider(coin_pair.second.out.scriptPubKey);
+        if (provider) {
+            providers.insert(std::move(provider));
+        }
+    }
+    if (providers.size() == 0) {
+        // When there are no available providers, use DUMMY_SIGNING_PROVIDER so we can check if the tx is complete
+        providers.insert(&DUMMY_SIGNING_PROVIDER);
+    }
+
+    UniValue result;
+    for (const SigningProvider* provider : providers) {
+        result = SignTransaction(mtx, provider, coins, request.params[2]);
```

Comment:
There shouldn't ever be a SigningProvider that doesn't match.

Reply:
In commit "Refactor: Require scriptPubKey to get wallet SigningProvider" (f7510c144b2262e1ee114eaa740b889c24ec7abb)

According to RPC documentation result has three fields: "hex" "complete" and "errors". I guess for "hex" and "complete" fields it makes sense to take the last values, but errors field should probably accumulate? In any case, this code is not very future proof in case new fields are added. Would suggest changing SignTransaction to take a `UniValue& result` input/output parameter, so it can be responsible for updating the result at each step, and this code isn't just blindly overwriting previous results.

Reply:
I've added a commit which changes `SignTransaction` to `void` and has a `result` parameter to accumulate the errors array.